x/epochs/depinject.go

```diff
@@ -39,7 +39,7 @@ type ModuleInputs struct {
 type ModuleOutputs struct {
 	depinject.Out
 
-	EpochKeeper keeper.Keeper
+	EpochKeeper *keeper.Keeper
```

Comment:
why does it need to be a pointer?

Reply:
This was a bug with x/epochs, in which hooks weren't being set. The issue was that we were doing `(k Keeper) SetHooks(...` instead of `(k *Keeper) SetHooks(...`. I copied what we do in x/gov to achieve the same